memcached.go

```diff
@@ -9,9 +9,7 @@ import (
 	"github.com/bradfitz/gomemcache/memcache"
 )
 
-var _ Client = &memcacheClient{}
```

Comment:
Why was this removed?

Reply:
Since the constructor now returns a `Client`, this is now superfluous.

Note that the usage doesn't change, so this shouldn't affect clients.